From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: GideonWhite1029 <gideonwhite1029@gmail.com>
Date: Wed, 1 May 2024 18:56:10 +0400
Subject: [PATCH] Information changed during startup


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 7196594e07af19a14c320d77df893978525fe386..9b02bc432bcd3fd89bb676693350ac0b17f83511 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -33,6 +33,13 @@ public interface ServerBuildInfo {
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
 
+    // PurpurAsync start
+    /**
+     * The brand id for PurpurAsync.
+     */
+    Key BRAND_PURPURASYNC_ID = Key.key("gideonwhite1029", "purpurasync");
+    // PurpurAsync end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 3a71f8fe73a2309f57cb06dedf204d3e37b46b03..0e2746869dedc1ab43ab1507cb1d3e62e5669707 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -14,6 +14,8 @@ import java.util.Set;
 import java.util.UUID;
 import java.util.function.Consumer;
 import java.util.logging.Logger;
+
+import io.papermc.paper.util.JarManifests;
 import org.bukkit.Warning.WarningState;
 import org.bukkit.advancement.Advancement;
 import org.bukkit.block.data.BlockData;
@@ -120,8 +122,28 @@ public final class Bukkit {
       */
     @NotNull
     public static String getVersionMessage() {
-        final io.papermc.paper.ServerBuildInfo version = io.papermc.paper.ServerBuildInfo.buildInfo();
-        return "This server is running " + getName() + " version " + version.asString(io.papermc.paper.ServerBuildInfo.StringRepresentation.VERSION_FULL) + " (Implementing API version " + getBukkitVersion() + ")";
+        final var manifest = JarManifests.manifest(Bukkit.getServer().getClass());
+        final String gitBranch = manifest == null ? null : manifest.getMainAttributes().getValue("Git-Branch");
+        final String gitCommit = manifest == null ? null : manifest.getMainAttributes().getValue("Git-Commit");
+        String branchMsg = " on " + gitBranch;
+        if ("master".equals(gitBranch) || "main".equals(gitBranch)) {
+            branchMsg = "";  // Don't show branch on main/master
+        }
+        // PurpurAsync start
+        return "\n" +
+                "\u001B[38;5;123m      ___                         ___                    \n" +
+                "\u001B[38;5;122m     / _ \\__ _________  __ ______/ _ | ___ __ _____  ____\n" +
+                "\u001B[38;5;121m    / ___/ // / __/ _ \\/ // / __/ __ |(_-</ // / _ \\/ __/\n" +
+                "\u001B[38;5;120m   /_/   \\_,_/_/ / .__/\\_,_/_/ /_/ |_/___/\\_, /_//_/\\__/ \n" +
+                "\u001B[38;5;119m                /_/                      /___/           \u001B[0m\n" +
+                "\n" +
+                "  \u001b[38;5;86m Server name: \u001B[0m" + getName() + "\n" +
+                "  \u001b[38;5;86m Version: \u001B[0m" + getVersion() + "\n" +
+                "  \u001b[38;5;86m API version: \u001B[0m" + getBukkitVersion() + "\n" +
+                "  \u001b[38;5;86m Github commit: \u001B[0m" + gitCommit + "\n" +
+                "\n";
+
+        // PurpurAsync stop
         // Paper end
     }
 
